server/controllers/StatusController.js

```diff
@@ -41,14 +41,16 @@ const getStatus = async (req, res) => {
       const mission = latestMission;
       const vehicle = await getVehicle(latestMission.vehicle_id);
       const status = 'in_mission';
-      const currentStatus = _.find(missionProgress, {'status': vehicle.status});
-      if (currentStatus.conditionForNextStatus(latestMission)){
-        const timestampString = currentStatus.nextMissionStatus + '_at';
-        let timestampObject = {};
-        timestampObject[timestampString] = Date.now();
-        await updateMission(latestMissionId, timestampObject);
-        await createMissionUpdate(latestMissionId, currentStatus.nextMissionStatus);
-        await updateVehicleStatus(latestMission.vehicle_id, currentStatus.nextVehicleStatus);
+      if (vehicle.status !== 'waiting_pickup'){
```

Comment:
This seems like an unsustainable way to approach this.

You'll have to keep adding conditions here for when to skip the check as mission flows get more complicated.

You've created an elegant structure in `missionProgress.js` that defines the flow, and here you are going around it for one specific case. Why not handle this status transition within `missionProgress.js` just like you handle `available`

Reply:
Good idea!